wl_consecutive_features.py

```diff
@@ -0,0 +1,126 @@
+        res = []
+        for i in range(0, len(col_names), 4):
+            time_period = col_names[i].split('_')[2]
+            if time_period == 'l5':
+                last_fights = 5
+            else:
+                last_fights = 0
+                        
+            has_diff = 'diff' in col_names[i]
+            fighter_a_cwins, fighter_a_closses, fighter_b_cwins, fighter_b_closses = self.compute_consecutive_wl_stats(df, fighter_a_id, fighter_b_id, index, last_fights, has_diff)
+            res.append(fighter_a_cwins)
+            res.append(fighter_a_closses)
+            res.append(fighter_b_cwins)
+            res.append(fighter_b_closses)
+                
+        return pd.Series(res)
+    
+    def compute_consecutive_wl_stats(self, df, fighter_a_id, fighter_b_id, index, last_fights, diff=False):
+        """
+        Computes the consecutive w/l stats for each fighter in the dataset
+
+        Parameters:
+        - df (pd.Dataframe): The original dataframe containing all the fights
+        - fighter_a_id (int): The id of the first fighter
+        - fighter_b_id (int): The id of the second fighter
+        - index (int): The index of the row in the dataframe
+        - col_names (list): The list of column names for the consecutive w/l stats
+
+        Returns:
+        - list: The list of consecutive w/l stats for each fighter
+        """
+        all_prev_fights = df.iloc[:index]
+        if not all_prev_fights.empty:
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            fighter_a_fights = all_prev_fights[(fighter_a_id_vals == fighter_a_id) | (fighter_b_id_vals == fighter_a_id)][-last_fights:]
+            fighter_b_fights = all_prev_fights[(fighter_b_id_vals == fighter_b_id) | (fighter_a_id_vals == fighter_b_id)][-last_fights:]
+
+            fighter_a_cwins = 0
+            fighter_a_closses = 0
+            fighter_b_cwins = 0
+            fighter_b_closses = 0
+
+            if not fighter_a_fights.empty:
+                for _, fight in fighter_a_fights.iterrows():
+                    if fight['winner_id'] == fighter_a_id:
+                        fighter_a_cwins += 1
+                        fighter_a_closses = 0
+                    else:
+                        fighter_a_closses += 1
+                        fighter_a_cwins = 0
+    
+            if not fighter_b_fights.empty:
+                for _, fight in fighter_b_fights.iterrows():
+                    if fight['winner_id'] == fighter_b_id:
+                        fighter_b_cwins += 1
+                        fighter_b_closses = 0
+                    else:
+                        fighter_b_closses += 1
+                        fighter_b_cwins = 0
+        
+            if diff:
+                return fighter_a_cwins - fighter_b_cwins, fighter_a_closses - fighter_b_closses, fighter_b_cwins - fighter_a_cwins, fighter_b_closses - fighter_a_closses
```

Comment:
Is it ok that the differential potentially could be negative or do bring it to zero if it is? In my opinion negative makes sense but thought I would mention it here

Reply:
Yes negative is fine